File: api-py/app/api/v1/tour.py
"""新手引导路由(对齐 NestJS tour.controller)。"""
from typing import Any, Dict

from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session

from app.api.deps import get_current_merchant
from app.core.response import success, success_response
from app.db.session import get_db
from app.services.tour import get_seen, mark_seen, reset_seen

router = APIRouter(prefix="/tour", tags=["新手引导"])


@router.get("/seen", summary="查询是否已看新手引导")
def seen(merchant: Dict[str, Any] = Depends(get_current_merchant), db: Session = Depends(get_db)):
    return success(get_seen(db, merchant["merchant_id"]))


@router.post("/seen", summary="标记已看", dependencies=[Depends(get_current_merchant)])
def seen_post(merchant: Dict[str, Any] = Depends(get_current_merchant), db: Session = Depends(get_db)):
    return success_response(mark_seen(db, merchant["merchant_id"]), status_code=200)


@router.post("/seen/reset", summary="重置为未看", dependencies=[Depends(get_current_merchant)])
def seen_reset(merchant: Dict[str, Any] = Depends(get_current_merchant), db: Session = Depends(get_db)):
    return success_response(reset_seen(db, merchant["merchant_id"]), status_code=200)

File: project/src/api/tour.ts
/**
 * 账号级新手引导标记接口（阶段一/二/数据专区共用）
 * @description GET /api/tour/seen 查询是否已看（data.seen）；POST /api/tour/seen 标记已看；
 *              POST /api/tour/seen/reset 重置（重新观看引导用）。
 */
import request from './request';

/** 查询引导是否已看（seen: true 已看 / false 未看） */
export const fetchTourSeen = () =>
  request.get<{ seen: boolean }>('/api/tour/seen').then((res) => res.data);

/** 标记引导已看（完成/跳过时调用） */
export const markTourSeen = () =>
  request.post<{ ok?: boolean }>('/api/tour/seen', {}).then((res) => res.data);

/** 重置引导已看（重新观看引导入口调用） */
export const resetTourSeen = () =>
  request.post<{ ok?: boolean }>('/api/tour/seen/reset', {}).then((res) => res.data);
